File: photomap-core-api/photo-api/src/main/java/ua/in/photomap/photoapi/dto/PhotoMetadataDTO.java
package ua.in.photomap.photoapi.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import ua.in.photomap.common.photo.model.dto.CoordinatesDTO;
import ua.in.photomap.common.photo.model.dto.YearRangeDTO;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class PhotoMetadataDTO {
    private String title;
    private String description;
    private String author;
    private String source;
    private YearRangeDTO yearRange;
    private Integer licenseId;
    private List<String> tags;
    private CoordinatesDTO coordinates;
}

File: photomap-core-api/photo-api-python/dto/photo_metadata_dto.py
from pydantic import BaseModel, Field
from typing import List, Optional
from dto.photo_dto import YearRangeDTO, CoordinatesDTO

class PhotoMetadataDTO(BaseModel):
    title: str = Field(description='Photo title')
    description: str = Field(description='Photo description')
    author: Optional[str] = Field(description='Photo author')
    source: str = Field(description='Photo source')
    yearRange: YearRangeDTO = Field(description='Year range when the photo was taken')
    licenseId: int = Field(description='License ID')
    tags: List[str] = Field(description='Photo tags')
    coordinates: CoordinatesDTO = Field(description='Photo coordinates') 